tests/unit/test_all_types.cpp

```diff
@@ -179,3 +179,34 @@ TEMPLATE_TEST_CASE("Scalar in std::vector<std::array>", "[Types]", bool, std::st
         CHECK(value.size() == 5);
     }
 }
+
+#if HIGHFIVE_CXX_STD >= 17
+TEMPLATE_PRODUCT_TEST_CASE("Scalar in std::vector<std::byte>", "[Types]", std::vector, std::byte) {
+    const std::string FILE_NAME("Test_vector_byte.h5");
+    const std::string DATASET_NAME("dset");
+    TestType t1(5, std::byte(0xCD));
+
+    {
+        // Create a new file using the default property lists.
+        File file(FILE_NAME, File::ReadWrite | File::Create | File::Truncate);
+
+        // Create the dataset
+        DataSet dataset = file.createDataSet(
+            DATASET_NAME, {5}, create_datatype<typename details::inspector<TestType>::base_type>());
+
+        // Write into the initial part of the dataset
+        dataset.write(t1);
```

Comment:
Despite the above tests not doing it, it might be nice to also check:
```
        auto dataset2 = file.createDataSet(DATASET_NAME2, t1);
```


We could leave this for when we refactor the whole file to be more DRY.

Reply:
ok, I leave this for the refactoring later.